public/app/plugins/panel/bargauge/module.tsx

```diff
@@ -61,19 +61,52 @@ export const plugin = new PanelPlugin<Options>(BarGaugePanel)
         defaultValue: defaultOptions.showUnfilled,
         showIf: (options) => options.displayMode !== 'lcd',
       })
-      .addNumberInput({
+      .addRadio({
+        path: 'sizing',
+        name: 'Bar size',
+        settings: {
+          options: [
+            { value: BarGaugeSizing.Auto, label: 'Auto' },
+            { value: BarGaugeSizing.Manual, label: 'Manual' },
+          ],
+        },
+        defaultValue: defaultOptions.sizing,
+      })
+      .addSliderInput({
         path: 'minVizWidth',
         name: 'Min width',
-        description: 'Minimum column width',
+        description: 'Minimum column width (vertical orientation)',
```

Comment:
@nmarrs There is one point to discuss.

I want to show specific sizing options for selected layout orientation. For example, to display `Min width` for vertical. layout orientation only, to avoid options overwhelming.

I didn't find any proper way to show/hide sizing options for specific layout orientations. For example, if there is `auto` layout orientation we can't say the exact orientation, so we can't decide which sizing options to show or hide. So right now it's gonna be displayed all sizing options both for vertical and horizontal layout.
Btw, the same situation happening with the `Name placement` option.

I decided to add an extra description to each sizing option. What do you think?

Reply:
I think this is reasonable :) I think this is good to show off to Torkel / get this thoughts / approval